pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -146,7 +146,10 @@ type PrometheusResponseExtractor struct{}
 
 // Extract extracts response for specific a range from a response.
 func (PrometheusResponseExtractor) Extract(start, end int64, from Response) Response {
```

Comment:
Should we expand the comment here to clarify whose responsibility it is to close `resp`, and how this relates to the returned `Response`? (eg. does closing `resp` also close the returned value?)

(same comment applies to `ResponseWithoutHeaders`)